DREAM/Optimization/tsgOptimizationUtils.hpp

```diff
@@ -71,6 +81,9 @@ inline void checkVarSize(const std::string method_name, const std::string var_na
     }
 }
 
+// Functions used in optimization.
+using GenericBatchedFunction = std::function<void(const std::vector<double> &x_batch, std::vector<double> &y_batch)>;
```

Comment:
Multiple functions have the same signature, but setting them all the the same `Generic` thing isn't helpful.
* the user has to look up the generic function to see what's going
* the user has to then read the documentation of the specific instance
* sometime `x_batch` is not actually a batch ... confusing

Here is a better approach:
* remove the generic function
* repeat the signature every time, but have different names for `x` and `y` as appropriate
```
ObjectiveFunction = std::function<void(const std::vector<double> &x_batch, std::vector<double> &fval_batch)>;
GradientFunctionSingle = std::function<void(const std::vector<double> &x_single, std::vector<double> &grad)>;
ProjectionFunctionSingle = std::function<void(const std::vector<double> &x_single, std::vector<double> &proj)>;
```
Have the documentation too.

Reply:
Changed. I have used the suggestions above.